### DISCUSSION TITLE
Add cost / price data and calculations for ingredients

### DISCUSSION THREAD

**Discussion by andrewvaughan** [mealie-recipes/mealie#2996] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

For ingredient data, it would be extremely useful to be able to record the cost/price per unit of an ingredient, and thus provide an estimated cost of a recipe (per-serving and per-cook) in each recipe.

There's even API integrations out there that might make this super easy for users (letting them use their own keys, of course). These could also be used to automatically look up nutrition information and improve that capabilitiy:
https://developer.edamam.com/food-database-api

Related - it seems to make sense to me that unit type would be a 1-to-1 (or, at minimum, 1-to-many) relationship with each ingredients, and not necessarily inputted at the recipe stage. If a recipe calls for a different unit (e.g., mass instead of volume, SI instead of Metric), these are all able to be converted using common data like densities and basic conversion math. I think an idea of "Convention over Configuration" would go a long way in allowing better integrations and capability rollout, such as price per unit, but I'm going down a rabbit hole.

### Please provide a concise description of the feature that would resolve your issue.

Let me tell you that bananas are $0.69 cents each and that the cream I buy is $4.20 per liter.

For advanced maths, incorporate an API that looks up the average cost of milk per gallon in the area I live. Fancy!

### Please consider and list out some caveats or tradeoffs made in your design decision

Allowing users to use any unit they wish at any time in the recipe entry screen gets messy.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by Kuchenpirat** [comment]
Hi, thanks for the interest in mealie. 
I see a few problematic items with the calculation of recipe pricing:

1. As we do currently not convert units you would have to provide cost for every unit of that particular food which is quite cumbersome.
2. Food pricing is fluctuating a lot (vegetables are seasonal etc) so this would have to be maintained quite a lot if we decide against the api route which would need us to do a lot to the current implementation of units. 
3. we would need to handle foods that are used without unit and foods that have not have an price added to them and display this to the user, because the estimated price can be way off.

With all this I don't see us implementing this.

**Comment by RealFoxie** [comment]
I have a custom script that does this and as mentioned above, a lot more plays a part in this than you'd at first realized. My own script took several weeks to get right.

A first step could be for unit conversions, but even just that step is not trivial

**Reply by ShamanTcler** [reply]
My wife does the kitchen at Church, so cost per serving is important. Is there a chance you would be willing to share the script?

**Comment by royaltongue** [comment]
Instead of price per unit, could it be implemented as price per package? If I have to use 3 carrots in a recipe, I'm less concerned over the price of those 3 carrots versus the price of the pack of 5 that I have to buy.

With this, the user could define whatever price they want for whatever package size they see fit. The big issue I see with this is for things like condiments and spices, which don't need to be bought for every recipe they're used in. As a workaround for this, there could be a setting like `Include in Price Calculations` that can be set within the Food Data
